scripts/translate_challenge.py

```
@@ -0,0 +1,261 @@
+    else:
+        xml_match = re.search(r"<xml>.*?</xml>", response, re.DOTALL)
+        if xml_match:
+            xml_content = xml_match.group(0)
+        else:
+            return "Failed to extract XML from response"
+
+    try:
+        # Use decode_friendly to parse XML
+        for element in decode_friendly(xml_content, tags="xml"):
+            return element
+        return "No XML element found after decoding"
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)